riptide-opentelemetry/src/test/java/org/zalando/riptide/opentelemetry/OpenTelemetryPluginTest.java

```diff
@@ -0,0 +1,252 @@
+        }
+
+        final List<SpanData> spans = otelTesting.getSpans();
+
+        assertThat(spans, hasSize(2));
+
+        assertThat(spans.get(1).getSpanId(), is(parent.getSpanContext().getSpanId()));
+
+        final SpanData child = spans.get(0);
+        assertThat(child.getParentSpanId(), is(parent.getSpanContext().getSpanId()));
+        assertThat(child.getStatus(), is(not(StatusData.error())));
+
+        final Attributes attributes = child.getAttributes();
+        assertThat(attributes.get(AttributeKey.stringKey("env")), is("unittest"));
+        assertThat(attributes.get(AttributeKey.stringKey("http.method")), is("POST"));
+        assertThat(attributes.get(AttributeKey.longKey("http.status_code")), is(200L));
+    }
+
+    @Test
+    void shouldTraceRequestAndServerError() {
+        driver.addExpectation(onRequestTo("/"), giveEmptyResponse().withStatus(500));
+
+        final Span parent = tracer.spanBuilder("test").startSpan();
+
+        try (final Scope ignored = parent.makeCurrent()) {
+            unit.get("/")
+                .call(pass())
+                .join();
+        } finally {
+            parent.end();
+        }
+
+        final List<SpanData> spans = otelTesting.getSpans();
+
+        assertThat(spans, hasSize(2));
+
+        assertThat(spans.get(1).getSpanId(), is(parent.getSpanContext().getSpanId()));
+
+        final SpanData child = spans.get(0);
+        assertThat(child.getParentSpanId(), is(parent.getSpanContext().getSpanId()));
+        assertThat(child.getStatus(), is(StatusData.error()));
+
+        final Attributes attributes = child.getAttributes();
+        assertThat(attributes.get(AttributeKey.stringKey("env")), is("unittest"));
+        assertThat(attributes.get(AttributeKey.stringKey("http.method")), is("GET"));
+        assertThat(attributes.get(AttributeKey.longKey("http.status_code")), is(500L));
+    }
+
+    @Test
+    void shouldTraceRequestAndNetworkError() {
+        driver.addExpectation(onRequestTo("/"), giveEmptyResponse().after(1, SECONDS));
+
+        final Span parent = tracer.spanBuilder("test").startSpan();
+
+        try (final Scope ignored = parent.makeCurrent()) {
+            final CompletableFuture<?> future = unit.get(URI.create(driver.getBaseUrl()))
+                                                    .call(noRoute());
+
+            final CompletionException error = assertThrows(CompletionException.class, future::join);
+            assertThat(error.getCause(), is(instanceOf(SocketTimeoutException.class)));
+        } finally {
+            parent.end();
+        }
+
+        final List<SpanData> spans = otelTesting.getSpans();
+
+        assertThat(spans, hasSize(2));
+
+        assertThat(spans.get(1).getSpanId(), is(parent.getSpanContext().getSpanId()));
+
+        final SpanData child = spans.get(0);
+        assertThat(child.getParentSpanId(), is(parent.getSpanContext().getSpanId()));
+        assertThat(child.getStatus(), is(StatusData.error()));
```

Comment:
Just for completeness, I'd recommend verifying that we're also adding details about the exception from the call to `recordException(...)` in the `ErrorSpanDecorator`
We can look for one of the attributes or events listed [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/exceptions.md).

Reply:
Added event verification